django/db/models/fields/__init__.py

```diff
@@ -270,16 +272,32 @@ def is_value(value, accept_promise=True):
                     for value, human_name in group_choices
                 ):
                     break
+                if self.max_length is not None:
+                    max_length = max(
+                        max_length,
+                        *(len(value) for value, human_name in group_choices if isinstance(value, str)),
+                    )
```

Comment:
I think we can add check for `group_choices` to the `if` condition and simplify `max()`, e.g.
```python
if self.max_length is not None and group_choices:
    max_length = max(
        len(value) for value, _ in group_choices if isinstance(value, str)
    )
```

Reply:
Same again here. (So yes to the `if` condition, but no to the simplification. Well I'll make it better, but we need the `max_choices_length = max(max_choices_length, *(...))` pattern.)